pkg/kubelet/kubeletconfig/util/codec/codec.go

```diff
@@ -26,9 +26,37 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig/scheme"
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig/v1beta1"
 )
 
-// TODO(mtaufen): allow an encoder to be injected into checkpoint objects at creation time? (then we could ultimately instantiate only one encoder)
+// EncodeKubeletConfig encodes an internal KubeletConfiguration to an external JSON representation
+func EncodeKubeletConfig(internal *kubeletconfig.KubeletConfiguration) ([]byte, error) {
```

Comment:
if this is moving to non-test code, do we need to allow specifying the target version here?

Reply:
It's not necessary yet. Maybe in the future, when multiple versions are available.

Reply:
> Maybe in the future, when multiple versions are available.

I don't want callers to silently get the version they persist changed out from under them... making the version written implicit leads to that.

Reply:
Fair enough, updated.